docs/guide_slash_and_interactions.rst

```diff
@@ -0,0 +1,297 @@
+            await interaction.response.send_message(f"Your number is {number}", ephemeral=True)
+
+See the `Discord.py documentation <https://discordpy.readthedocs.io/en/stable/interactions/api.html#range>`__ for more information on this.
+
+
+---------------
+Hybrid Commands
+---------------
+Hybrid commands are a way to bridge the gap between text commands and slash commands.
+These types of commands allow you to write a text and slash command simultaneously using the same function.
+This is useful for commands that you want to be able to use in both text and slash commands.
+
+.. note::
+    As with slash command groups, top level hybrid command groups **cannot** be invoked as a slash command. They can however be invoked as a text command.
+
+.. code-block:: python
+
+    from redbot.core import commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @commands.hybrid_command(name="cat")
+        async def cat(self, ctx: commands.Context):
+            await ctx.send("Meow")
+
+        @commands.hybrid_group(name="dog")
+        async def dog(self, ctx: commands.Context):
+            await ctx.send("Woof")
+            # As discussed above, top level hybrid command groups cannot be invoked as a slash command.
+            # Thus, this will not work as a slash command.
+
+        @dog.command(name="bark")
+        async def bark(self, ctx: commands.Context):
+            await ctx.send("Bark")
+
+After syncing your cog via the :ref:`Slash<core-command-slash>` command, you'll be able to use the commands as both a slash and text command.
+
+---------------------
+Context Menu Commands
+---------------------
+Context menu commands are a way to provide a interaction via the context menu.
+These are seen under ``Apps`` in the Discord client when you right click on a message or user.
+Context menu commands are a great way to provide a quick way to interact with your bot.
+So long your command doesn't require any arguments that is.
+As these commands do not have the ability to accept any arguments.
+
+Setting up context commands is a bit more involved then setting up slash commands.
+First lets setup our context commands in our cog.
+
+.. code-block:: python
+    
+    import discord
+
+    from redbot.core import commands, app_commands
+
+
+    # Important: we're building the commands outside of our cog class.
+    @app_commands.context_menu(name="Get message ID")
+    async def get_message_id(interaction: discord.Interaction, message: discord.Message):
+        await interaction.response.send_message(f"Message ID: {message.id}", ephemeral=True)
+
+    @app_commands.context_menu(name="Get user ID")
+    async def get_user_id(interaction: discord.Interaction, user: discord.User):
+        await interaction.response.send_message(f"User ID: {user.id}", ephemeral=True)
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def teardown(bot):
```

Comment:
If teardown is here, and not in `__init__.py`, it needs to be imported. Better yet, it too should be in `__init__.py`.